lib/Parser/Parser.yy

```diff
@@ -632,23 +810,53 @@ unique_shared_opt:
 
 literal:
       boolean_literal
-    | intLit
-    | floatLit
-    | stringLit
+    | intLit { 
+          int intVal = std::stoi($1.getLexeme().str());
```

Comment:
Use StringRef::getAsInteger to create an APInt, it will automatically determine the amount of bits needed, instead of not supporting ints bigger than 32 bits...